ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+    // create a query based on the current visible extent
+    ogcFeatureExtentQueryParameters.setGeometry(datasetExtent);
+    // set a limit of 3000 on the number of returned features per request, the default on some services could be as low as 10
+    ogcFeatureExtentQueryParameters.setMaxFeatures(3000);
+
+    try {
+      // populate and load the table with the query parameters
+      // set the clearCache parameter to true to clear existing table entries, set the outfields parameter to null to request all fields
+      ogcFeatureCollectionTable.populateFromServiceAsync(ogcFeatureExtentQueryParameters, true, null).addDoneListener(() -> {
+        progressIndicator.setVisible(false);
+        applyQueryButton.setDisable(false);
+        // display number of features returned
+        setFeatureNumberLabelText();
+      });
+
+      // handle UI
+      revertToInitialQueryButton.setDisable(true);
+      textField.setText("3000");
+      // reset combo box and show original prompt text
+      comboBox.getSelectionModel().clearSelection();
+      comboBox.setButtonCell(new ListCell<>() {
+        @Override
+        protected void updateItem(String item, boolean empty) {
+          setText(item);
+        }
+      });
+
+    } catch (Exception exception) {
+      exception.printStackTrace();
+      new Alert(Alert.AlertType.ERROR, exception.getMessage()).show();
+    }
+  }
+
+  /**
+   * Populates features from provided query parameters, and displays the result on the map.
+   */
+  @FXML
+  private void query() {
+
+    QueryParameters cqlQueryParameters = new QueryParameters();
+    // set the query parameter's where clause with the CQL query in the combo box
+    cqlQueryParameters.setWhereClause(comboBox.getSelectionModel().getSelectedItem());
+    // set the max features to the number entered in the text field
+    cqlQueryParameters.setMaxFeatures(Integer.parseInt(textField.getText()));
+
+    // if the time extent checkbox is selected, retrieve the date selected from the date picker and set it to the
+    // query parameters time extent
+    if (timeExtentCheckBox.isSelected()) {
+      // get the value selected from the date picker
+      LocalDate startDate = startDatePicker.getValue();
+      LocalDate endDate = endDatePicker.getValue();
+
+      // convert the value into a calendar
+      Calendar start = new Calendar.Builder().setDate(startDate.getYear(), startDate.getMonthValue(), startDate.getDayOfMonth()).build();
+      Calendar end = new Calendar.Builder().setDate(endDate.getYear(), endDate.getMonthValue(), endDate.getDayOfMonth()).build();
+
+      // set the query parameters time extent
+      cqlQueryParameters.setTimeExtent(new TimeExtent(start, end));
+    }
+
+    // populate and load the table with the query parameters
+    // set the clearCache parameter to true to clear existing table entries and set the outfields parameter to null requests all fields
```

Comment:
>set the outfields parameter to null requests all fields

Something isn't quite right. Maybe change to "to null **in order to** request all fields"?